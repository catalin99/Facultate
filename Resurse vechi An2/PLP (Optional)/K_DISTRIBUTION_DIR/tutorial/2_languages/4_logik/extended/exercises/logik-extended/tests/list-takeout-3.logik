takeout(X,[X|R],R).
takeout(X,[F|R],[F|S]) :- takeout(X,R,S).

// 4 solutions
?- takeout(X,L,[1,2,3]).
