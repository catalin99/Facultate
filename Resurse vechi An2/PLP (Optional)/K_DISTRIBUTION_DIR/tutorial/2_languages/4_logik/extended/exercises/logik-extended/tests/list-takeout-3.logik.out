Search results:

Solution 1:
Solution:Map -->
"L" |-> [ 1 | ([ 2 | ([ 3 | ([ X | ([ .List{","} ]) ]) ]) ]) ]
"X" |-> X

Solution 2:
Solution:Map -->
"L" |-> [ 1 | ([ 2 | ([ X | ([ 3 | ([ .List{","} ]) ]) ]) ]) ]
"X" |-> X

Solution 3:
Solution:Map -->
"L" |-> [ 1 | ([ X | ([ 2 | ([ 3 | ([ .List{","} ]) ]) ]) ]) ]
"X" |-> X

Solution 4:
Solution:Map -->
"L" |-> [ X | ([ 1 | ([ 2 | ([ 3 | ([ .List{","} ]) ]) ]) ]) ]
"X" |-> X
